### PR TITLE
[MPS] Support SymInt captures in flex_attention score_mod/mask_mod

### PR DESCRIPTION
**Pull request by Isalia20**
Should be merged after #188362 is merged and branch is rebased.
i.e. this works after this PR:
```
import torch
from torch.nn.attention.flex_attention import create_block_mask, flex_attention

def run(q, k, v):
    limit = (q.size(-2) // 32 + k.size(-2) // 32) * 32
    block_mask = create_block_mask(
        lambda b, h, q_idx, kv_idx: q_idx < limit,
        B=None, H=None, Q_LEN=q.size(-2), KV_LEN=k.size(-2), device="mps",
    )
    return flex_attention(q, k, v, block_mask=block_mask)

q = torch.randn(1, 2, 192, 32, device="mps", dtype=torch.bfloat16)
k = torch.randn(1, 2, 128, 32, device="mps", dtype=torch.bfloat16)
v = torch.randn(1, 2, 128, 32, device="mps", dtype=torch.bfloat16)
compiled = torch.compile(run, fullgraph=True, dynamic=True)
torch.testing.assert_close(compiled(q, k, v), run(q, k, v), atol=5e-3, rtol=1e-3)
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/codegen/metal_flex_attention_template.py [hunk 1]
```diff
@@ -36,6 +37,27 @@
...
+# downstream floordiv/index stays exact; otherwise it falls back to float.
+_INT_PRESERVING_OPS = {operator.add: "+", operator.sub: "-", operator.mul: "*"}
+_CMP_OPS = {
+    operator.lt: "<",
+    operator.le: "<=",
+    operator.gt: ">",
+    operator.ge: ">=",
+    operator.eq: "==",
+    operator.ne: "!=",
+}
```
Comment (malfet) [inline torch/_inductor/codegen/metal_flex_attention_template.py:59]: Again, I have a feeling it all can be copy-n-pasted from metal kernel or some other abstractions

File: torch/_inductor/codegen/metal_flex_attention_template.py [hunk 2]
```diff
@@ -224,7 +261,48 @@ def _val(node) -> str:
...
+            )
+        elif target in (operator.floordiv, operator.mod):
+            both_int = _is_int(args[0]) and _is_int(args[1])
+            fa = f"static_cast<float>({_val(args[0])})"
+            fb = f"static_cast<float>({_val(args[1])})"
+            if target is operator.floordiv:
+                expr = f"metal::floor({fa} / {fb})"
+            else:
+                expr = f"c10::metal::remainder({fa}, {fb})"
+            ctype = "long" if both_int else "float"
```
Comment (malfet) [inline torch/_inductor/codegen/metal_flex_attention_template.py]: IMO this is wrong, if you are using it to construct some offset etc, then performing floating division will result in a wrong precision, you should just do `args[0] / args[1]`

Comment (malfet) [reply torch/_inductor/codegen/metal_flex_attention_template.py]: Example (suggested by Claude: 16777217 / 1 = 16777217, but 16777217.0 / 1.0 = 16777216

**Review by malfet — APPROVED** [review]
Probably LGTM, but see comment about floating devision accuracy

File: torch/_inductor/codegen/metal_flex_attention_template.py (hunk 2 above)
Comment (Isalia20) [reply torch/_inductor/codegen/metal_flex_attention_template.py]: fixed

File: torch/_inductor/codegen/metal_flex_attention_template.py (hunk 1 above)
Comment (Isalia20) [reply torch/_inductor/codegen/metal_flex_attention_template.py:59]: Comparisons could be folded in OP_TABLE but the arithmetic needs integer typing which I think OP_TABLE doesn't provide for SymInts.
